src/log/log.c

```diff
@@ -2227,30 +2226,21 @@ __log_write_internal(WT_SESSION_IMPL *session, WT_ITEM *record, WT_LSN *lsnp,
 		if (free_slot)
 			__wt_log_slot_free(session, myslot.slot);
 	} else if (force) {
-		/*
-		 * If we are going to wait for this slot to get written,
-		 * signal the wrlsn thread.
-		 *
-		 * XXX I've seen times when conditions are NULL.
-		 */
-		if (conn->log_cond != NULL) {
-			__wt_cond_signal(session, conn->log_cond);
-			__wt_yield();
-		} else
-			WT_ERR(__wt_log_force_write(session, 1, NULL));
+		__wt_cond_signal(session, &conn->log_cond);
+		__wt_yield();
```

Comment:
I sat with @sulabhM and tried to work through this code.  We could add a boolean to track whether `__wt_logmgr_create` has been called, but it seems weird that recovery can call into here without the logging being initialized.

While on the topic, having `reconfigure` call `__wt_logmgr_create` seems kind of backwards to me.  Elsewhere we call `*_create` during `wiredtiger_open` and `*_destroy` during `WT_CONNECTION::close`.

I don't see any harm in unconditionally initializing the condition variable and keeping it live until the connection is closed.  In other words, I don't think there's any reason to destroy and recreate the condition variable as part of a `reconfigure`.

Would it be reasonable to separate out "initializing" logging (done early, before any calls through here) from the current `__wt_logmgr_open`, assuming we can't move all of open before recovery?